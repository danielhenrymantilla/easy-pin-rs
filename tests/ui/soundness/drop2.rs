#![allow(unused)]

use ::easy_pin::{
    easy_pin,
    PinSensitive,
};
use ::core::ptr::NonNull;

#[easy_pin(Drop)]
pub
struct SelfReferential {
    #[transitively_pinned]
    string: PinSensitive<String>,

    #[unpinned]
    at_string: NonNull<String>,
}

impl Drop for SelfReferential {
    fn drop (self: &mut Self) {
        ::core::mem::replace(
            &mut self.string,
            PinSensitive::new("".into()),
        );
    }
}

fn main ()
{}
