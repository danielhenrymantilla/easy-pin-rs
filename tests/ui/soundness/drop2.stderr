error[E0119]: conflicting implementations of trait `std::ops::Drop` for type `SelfReferential`:
  --> $DIR/drop2.rs:9:12
   |
9  | #[easy_pin(Drop)]
   |            ^^^^ conflicting implementation for `SelfReferential`
...
19 | impl Drop for SelfReferential {
   | ----------------------------- first implementation here

For more information about this error, try `rustc --explain E0119`.
